sciter/event.py

```diff
@@ -16,7 +16,7 @@ class EventHandler:
     ALL_EVENTS = EVENT_GROUPS.HANDLE_ALL
     DEFAULT_EVENTS = EVENT_GROUPS.HANDLE_INITIALIZATION | EVENT_GROUPS.HANDLE_SIZE | EVENT_GROUPS.HANDLE_BEHAVIOR_EVENT | EVENT_GROUPS.HANDLE_SCRIPTING_METHOD_CALL
 
-    def __init__(self, window=None, element=None, subscription=DEFAULT_EVENTS):
+    def __init__(self, window=None, element=None, subscription=DEFAULT_EVENTS, script_call_exception_handler=None):
```

Comment:
We should document the new parameter.

Also, I'm not sure passing it as a ctor's argument is the most convenient way. Perhaps, a separate method? Or an overridable event? I need to play with it and think.

Reply:
I was actually thinking about doing something like that before I submitted the pull request. I'm not sure why I decided against it because I think from a cleanliness standpoint, having a class method that can be overridden to modify the behavior would be much cleaner, as you said. So tacking on to your pointer above about being able to intercept the returned exception, what do you think about adding something like this?

```python
class EventHandler:
    
    ...

    def _on_script_call(self, f):
        # update handlers on every call if needed
        self._dispatcher_update()

        fname = f.name.decode('utf-8')
        fn = self._dispatcher['handlers'].get(fname)
        rv = None

        # call raw handler first
        if True:
            try:
                args = [sciter.Value(f.argv[i]) for i in range(f.argc)]
                # pylint: disable=assignment-from-none,assignment-from-no-return
                # because the `self.on_` methods can be overloaded
                rv = self.on_script_call(fname, args)
            except Exception as e:
                import traceback
                traceback.print_exc()
                rv = self.script_exception_handler(fname, e)

        # if not handled, call decorated method
        if rv is None and fn:
            cfg = getattr(fn, '_sciter_cfg', {})
            skip_exception = not cfg.get('safe', True)
            try:
                if cfg.get('convert'):
                    args = sciter.Value.unpack_from(f.argv, f.argc)
                rv = fn(*args)
            except Exception as e:
                import traceback
                traceback.print_exc()
                exc = self.script_exception_handler(fname, e)
                rv = str(exc) if skip_exception else exc

        # if handled, pack result for Sciter
        if fn or rv is not None:
            sciter.Value.pack_to(f.result, rv)
            return True
        return False


    ...
    

    def script_exception_handler(self, func_name, exception):
        """
        By default, just return passed in exception.
        Can be overridden to change script exception handling.
        """
        return exception
```

Reply:
Yep. Now it looks perfect!